posts/2024-05-07-24.0.0.5-beta.adoc

```diff
@@ -0,0 +1,554 @@
+
+=== Changes to EE integration
+
+The parts of the CDI specification that define how CDI integrates with other Jakarta EE technologies moved into the Jakarta EE 11 Platform, Web Profile, and Core Profile specifications. This has no effect on users, but if you use the link:https://jakarta.ee/specifications/cdi/4.1/jakarta-cdi-spec-4.1[CDI specification] as a documentation reference, you will find that some parts of it are now in the link:https://jakarta.ee/specifications/platform/11/[Plaform], link:https://jakarta.ee/specifications/webprofile/11/[Web Profile] and link:https://jakarta.ee/specifications/coreprofile/11/[Core Profile] specifications instead.
+
+To use the CDI 4.1 feature, add `cdi-4.1` to your `server.xml` file:
+
+[source,xml]
+----
+<featureManager>
+  <feature>cdi-4.1</feature>
+</featureManager>
+----
+
+Further references:
+
+* link:https://jakarta.ee/specifications/cdi/4.1/apidocs/[CDI API]
+* link:https://jakarta.ee/specifications/interceptors/2.2/apidocs/[Interceptors API]
+* link:https://jakarta.ee/specifications/annotations/3.0/apidocs/jakarta.annotation/module-summary.html[Jakarta Annotations API]
+* link:https://jakarta.ee/specifications/cdi/4.1/jakarta-cdi-spec-4.1[CDI Specification]
+* link:https://jakarta.ee/specifications/interceptors/2.2/jakarta-interceptors-spec-2.2[Interceptors Specification]
+* link:https://jakarta.ee/specifications/annotations/3.0/annotations-spec-3.0[Jakarta Annotations Specification]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28075
+// Contact/Reviewer: mswatosh
+// // // // // // // //
+[#concurrent]
+== Jakarta Concurrency 3.1
+
+Jakarta Concurrency allows applications to use concurrency while maintaining the benefits of running on a Jakarta EE runtime. Jakarta Concurrency 3.1 is the new version for the upcoming Jakarta EE 11, and it adds support for new features of the latest Java SE releases as well as some usability improvements.
+
+The new features in Jakarta Concurrency 3.1 are improved support for the Java Flow (Reactive Streams) APIs, and the use of virtual threads in `ManagedExecutors` and `ManagedThreadFactories`. Support is also included for using `@Inject` in place of `@Resource`, and the option to run asynchronous methods on a schedule by using the new `@Schedule` annotation.
+
+To support the Flow APIs, `ContextService` has two new methods, `contextualSubscriber`, which provides context in a `Flow.Subscriber`, and `Flow.Processor`, which provides context in a `contextualProcessor`. These new methods allow for context in a flow when using a `Flow.Publisher` which does not allow configuring a `ManagedExecutor`.

```

Comment:
The second method name/class name combination is backwards.
```suggestion
To support the Flow APIs, `ContextService` has two new methods, `contextualSubscriber`, which provides context in a `Flow.Subscriber`, and `contextualProcessor`, which provides context in a `Flow.Processor`. These new methods allow for context in a flow when using a `Flow.Publisher` that does not allow configuring a `ManagedExecutor`.

```

Reply:
sorry, missed this. Fixed now.